tests/spread/store/resources/task.yaml

```
@@ -64,11 +64,11 @@ execute: |
 
   # upload and verify the file resource
   upload_revision_number=$(charmcraft upload-resource $CHARM_DEFAULT_NAME example-file --format=json --file=resourcefile.txt --arch=amd64,riscv64 | jq -r '.revision')
-  revisions_json=$(charmcraft resource-revisions $CHARM_DEFAULT_NAME example-file --format=json)
-  upload_revision=$(echo "$revisions_json" | jq -c ".[] | select(.revision == $upload_revision_number)" )
+  charmcraft resource-revisions $CHARM_DEFAULT_NAME example-file --format=json > file_revisions.json
+  upload_revision=$(cat file_revisions.json | jq -c ".[] | select(.revision == $upload_revision_number)" )
   upload_revision_architectures=$(echo "$upload_revision" | jq -c '.bases[0].architectures')
   [[ "$upload_revision_architectures" = '["amd64","riscv64"]' ]]
-  last_revision=$(echo "$revisions_json" | jq -r .[-1])
+  last_revision=$(cat file_revisions.json | jq -r .[-1])
```

I believe you can use jq directly on file without piping
```suggestion
  last_revision=$(jq -r .[-1] file_revisions.json)
```

---

Thanks! Add me here: https://porkmail.org/era/unix/award